### PR TITLE
Make the prometheus charm use a better Grafana interface

### PR DESCRIPTION
**Pull request by rbarry82**
Update the prometheus charm to use smarter data bags instead of dumping a raw dict into the relation

### CODE REVIEW

**Review by balbirthomas — APPROVED** [review]
Just some clarifications otherwise LGTM.

**Review by Abuelodelanada — APPROVED** [review]
LGTM

File: src/charm.py [hunk 1]
```diff
@@ -114,17 +116,25 @@ def _on_stop(self, _):
...
+
+        source_data = {
+            "private-address": str(
+                self.model.get_binding(event.relation).network.bind_address
+            ),
+            "port": str(self.model.config["port"]),
+            "source-type": "prometheus"
+        }
+
+        event.relation.data[self.unit]["sources"] = json.dumps(source_data)
```
[the remark this answers is not in the document]
Comment (mmanciop) [reply src/charm.py:137]: Grafana can have multiple datasources fetch stuff from the same prometheus

[the remark this answers is not in the document]
Comment (balbirthomas) [reply src/charm.py:137]: As I understand this stores a *single* bind address on the unit data bag for the leader *unit* only.

**Pull request closed without merging**